src/charm.py

```diff
@@ -187,18 +229,20 @@ def configure_workload_container(self, event) -> None:
                 logger.info(msg)
                 self.unit.status = MaintenanceStatus(msg)
                 container.add_layer(CONTAINER_NAME, pebble_config, combine=True)
+                container.pebble.replan_services()
 
-                service = container.get_service(CONTAINER_NAME)
-                if service.is_running():
-                    msg = "Stopping content-cache"
-                    logger.info(msg)
-                    self.unit.status = MaintenanceStatus(msg)
-                    container.stop(CONTAINER_NAME)
-
-                msg = "Starting content-cache"
+            exporter_container = self.unit.get_container(EXPORTER_CONTAINER_NAME)
+            if exporter_container.can_connect():
+                msg = "Updating exporter pebble layer config"
                 logger.info(msg)
                 self.unit.status = MaintenanceStatus(msg)
-                container.start(CONTAINER_NAME)
+                exporter_container.add_layer(
+                    EXPORTER_CONTAINER_NAME, exporter_config, combine=True
+                )
+                exporter_container.pebble.replan_services()
+            else:
+                self.unit.status = WaitingStatus("waiting for Pebble in workload container")
+
         except ConnectionError:
```

Comment:
It would be better to check if `container.can_connect` instead of catching an exception here